ds3-contract-comparator-print/src/main/java/com/spectralogic/ds3contractcomparator/print/htmlprinter/generators/row/AddedHtmlRowGenerator.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * ******************************************************************************
+ *   Copyright 2016-2017 Spectra Logic Corporation. All Rights Reserved.
+ *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *   this file except in compliance with the License. A copy of the License is located at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file.
+ *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ****************************************************************************
+ */
+
+package com.spectralogic.ds3contractcomparator.print.htmlprinter.generators.row;
+
+import com.google.common.collect.ImmutableList;
+import com.spectralogic.ds3contractcomparator.print.htmlprinter.models.body.rows.AddedRow;
+import com.spectralogic.ds3contractcomparator.print.htmlprinter.models.body.rows.NoChangeRow;
+import com.spectralogic.ds3contractcomparator.print.htmlprinter.models.body.rows.Row;
+
+import java.lang.reflect.Field;
+import java.util.Optional;
+
+import static com.spectralogic.ds3contractcomparator.print.utils.HtmlRowGeneratorUtils.getListPropertyFromObject;
+import static com.spectralogic.ds3contractcomparator.print.utils.HtmlRowGeneratorUtils.getPropertyValue;
+import static com.spectralogic.ds3contractcomparator.print.utils.HtmlRowGeneratorUtils.toFieldIndent;
+
+/**
+ * Utilities for generating the {@link ImmutableList} of {@link Row} which
+ * represent added Ds3 Objects using reflection.
+ */
+public final class AddedHtmlRowGenerator {
+
+    private AddedHtmlRowGenerator() {
+        //pass
+    }
+
+    /**
+     * Recursively traverse an object using reflection and construct the added
+     * rows that represent the object.
+     */
+    public static <T> ImmutableList<Row> createAddedRows(final T object, final int indent) {
```

Comment:
I think they can be, it would require using a lambda for the `createAddedRows` and `createDeletedRows`

So you would split the logic into a separate static method that takes the same set of arguments, but the added 'operation' lambda.